data/netcdf_data.py

```diff
@@ -757,34 +757,46 @@ def get_nc_file_list(self, datasetconfig: DatasetConfig, **kwargs: dict) -> Unio
             except KeyError:
                 raise RuntimeError(
                     "Opening a dataset via sqlite requires the 'variable' keyword argument.")
-            if not isinstance(variable, list):
-                variable = [variable]
+            if not isinstance(variable, set):
+                variable = set(variable)
 
-            calculated_variables = datasetconfig.calculated_variables
-            if variable[0] in calculated_variables:
-                equation = calculated_variables[variable[0]]['equation']
-
-                variable = data.utils.get_data_vars_from_equation(
-                    equation, [v.key for v in db.get_data_variables()])
```

Comment:
I stumbled across this issue a few days ago when I noticed that CTD profiles and T/S diagrams weren't working. Sorry, I didn't mention it, or provide a patch, but here is the solution I came up with:
```python
            if not isinstance(variable, list):
                variable = set([variable])
            else:
                variable = set(variable)

            calculated_variables = datasetconfig.calculated_variables
            for var in variable:
                if var in calculated_variables:
                    equation = calculated_variables[var]['equation']
                    equation_vars = data.utils.get_data_vars_from_equation(
                            equation, [v.key for v in db.get_data_variables()])
                    variable.union(set(equation_vars))
            variable = list(variable)
```
More casting back and forth between `list` and `set` than I would like - perhaps some can be polished. Using `set` to avoid duplication of vars from eqns, IIRC.

Grist for the mill...

Reply:
No worries! It's a reassurance for me that my solution is similar in thinking to yours :1st_place_medal:
To avoid the `variable.union(set(equation_vars))` cast,  I used a `variable.update(equation_vars)` which grabs stuff from other iterables.